src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/snapshot/AttemptSnapshotStep.kt

```diff
@@ -0,0 +1,75 @@
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.step.snapshot
+
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.elasticapi.suspendUntil
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.ManagedIndexMetaData
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.SnapshotActionConfig
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.managedindexmetadata.StepMetaData
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.step.Step
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.action.admin.cluster.snapshots.create.CreateSnapshotRequest
+import org.elasticsearch.action.admin.cluster.snapshots.create.CreateSnapshotResponse
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.service.ClusterService
+import org.elasticsearch.rest.RestStatus
+
+class AttemptSnapshotStep(
+    val clusterService: ClusterService,
+    val client: Client,
+    val config: SnapshotActionConfig,
+    managedIndexMetaData: ManagedIndexMetaData
+) : Step(name, managedIndexMetaData) {
+
+    private val logger = LogManager.getLogger(javaClass)
+    private var stepStatus = StepStatus.STARTING
+    private var info: Map<String, Any>? = null
+
+    @Suppress("TooGenericExceptionCaught")
+    override suspend fun execute() {
+        try {
+            logger.info("Executing snapshot on ${managedIndexMetaData.index}")
+            val createSnapshotRequest = CreateSnapshotRequest()
+                .indices(managedIndexMetaData.index)
+                .snapshot(config.snapshot)
+                .repository(config.repository)
+            if (config.includeGlobalState != null) {
+                createSnapshotRequest.includeGlobalState(config.includeGlobalState)
```

Comment:
I've changed the logic

Reply:
I did some testing with `include_global_state` and it's not something we can include at this point in time.

It's including our custom index metadata in the snapshot so when you restore it actually resets the index being managed back to that point in time when the snapshot occurred.

@dabr-grapeup @JavierAbrego Could you remove `include_global_state` from the steps/action/parsing so it's not included for now. This can be a followup on how we can include this in the snapshot action. Either we need to find a way to make sure the ISM custom index metadata is not included in the global state with the snapshot or we need to consider moving the metadata out of the cluster state and possibly into the config index.